native/src/layout/flex.rs

```diff
@@ -124,19 +145,34 @@ where
                 max_main
             };
 
-            let (min_main, min_cross) =
-                axis.pack(min_main, axis.cross(limits.min()));
+            let (min_main, min_cross) = axis.pack(
+                min_main,
+                if align_items == Align::Fill {
+                    cross
+                } else {
+                    axis.cross(limits.min())
+                },
+            );
 
-            let (max_main, max_cross) =
-                axis.pack(max_main, axis.cross(limits.max()));
+            let (max_main, max_cross) = axis.pack(
+                max_main,
+                if align_items == Align::Fill {
+                    cross
+                } else {
+                    axis.cross(limits.max())
+                },
+            );
```

Comment:
With `Align::Fill`, I think there's an inconsistency in how the `child.layout` is being calculated for children with and without fill factor.

In the first iteration, I get the max `cross` of all non-fill on cross axis children. But the nodes on `fill_factor == 0` that are stored aren't bounded by that max `cross` that's calculated. However, I'm bounding the `fill_factor != 0` nodes by this max `cross`.

I think I need to add a new iteration to calculate `cross` up front, that I can then bound ALL nodes by.

Reply:
Fixed in 7631c70